presto-tests/src/main/java/com/facebook/presto/tests/datatype/DataType.java

```diff
@@ -130,9 +130,14 @@
                 bigDecimal -> bigDecimal);
     }
 
-    private static String quote(String value)
+    public static String quote(String value)
     {
-        return "'" + value + "'";
+        return "'" + value.replace("'", "''") + "'";
+    }
+
+    public static String quotedString(Object value)
```

Comment:
Quite crude. Is `quotedString(new BigDecimal("1.0"))` a valid literal of corresponding (decimal) type? I think it's not. I'r rather have `dataType("date", DateType.DATE, DataType::quotedString, identity())`, `dataType("time", TimeType.TIME, DataType::quotedString, identity())` wrapped in named method (`dateDataType`, `timeDataType`). 
(If you're concerned those types postgres specific (I don't think so), they could be defined in the postgres test class.)

Reply:
Agreed. Removed all this for now since it's not needed.